src/runtime/primitives/system_call.zig

```diff
@@ -148,7 +148,7 @@ pub fn Write_BytesFrom_AtOffset_Into_IfFail(context: *PrimitiveContext) !Executi
     const arguments = context.getArguments("_Write:BytesFrom:AtOffset:Into:IfFail:");
     const bytes_to_write = try arguments.getInteger(0, .Unsigned);
     const byte_array = try arguments.getObject(1, .ByteArray);
-    const offset = try arguments.getInteger(2, .Unsigned);
+    const offset: usize = @intCast(try arguments.getInteger(2, .Unsigned));
```

Comment:
Can we add a bounds check to this? This will crash the compiler in safe release modes if the `AtOffset:` argument receives a value greater than `std.math.maxInt(usize)`.

Reply:
What would the expected behavior be when the value is too big? Should I return an error in that case?

Reply:
Yup. I think we're sloppy about this in other places too, so having a utility which helps us bounds check these values would be nice:

```zig
pub fn exceedsBoundsOf(value: anytype, comptime T: type) bool {
    const value_type_info = @typeInfo(@TypeOf(value));
    const target_type_info = @typeInfo(T);

    if (value_type_info.Int.signedness != target_type_info.Int.signedness)
        @compileError("Can't bounds check for types of different signedness");

    if (value_type_info.Int.bits < target_type_info.Int.bits)
        return false;

    // Prevent triggering UB ourselves by using the largest type between the two.
    const PeerType = @TypeOf(value, @as(T, undefined));

    if (@as(PeerType, value) > @as(PeerType, std.math.maxInt(T)))
        return true;

    // The check above is sufficient for unsigned values, but not signed ones.
    if (value_type_info.Int.signedness != .signed)
        return false;

    if (@as(PeerType, value) < @as(PeerType, std.math.minInt(T)))
        return true;

    return false;
}
```

You can then use this like `exceedsBoundsOf(offset, usize)` and safely `@intCast` away in code below. For how to return an error, search for `Completion.initError` in primitives.

Reply:
I added the utility and the check.